components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/sync/WorkManagerSyncManager.kt

```diff
@@ -372,14 +394,24 @@ internal class WorkManagerSyncWorker(
         // to implement/reason about than worker reconfiguration.
         val deviceSettings = FxaDeviceSettingsCache(context).getCached()!!
 
+        // Obtain encryption keys for stores that came along with KeyProviders.
+        // This can take a bit of time!
+        val localEncryptionKeys = engineKeyProviders.mapKeys {
+            it.key.nativeName
+        }.mapValues {
+            // At this point, we don't really care how we got the key, so ignore KeyGenerationReason.
```

Comment:
this is wrong actually, we do care here and need to invoke recovery APIs on the storage once they're in place...